crates/rolldown_dev/tests/hmr_full_bundle_mode/continuous_generate_hmr_patch/artifacts.snap

```diff
@@ -0,0 +1,150 @@
+---
+source: crates/rolldown_testing/src/integration_test.rs
+---
+# Assets
+
+## main.js
+
+```js
+// HIDDEN [rolldown:runtime]
+// HIDDEN [rolldown:hmr]
+//#region hmr.js
+var hmr_exports = /* @__PURE__ */ __export({ foo: () => foo });
+const hmr_hot = __rolldown_runtime__.createModuleHotContext("hmr.js");
+__rolldown_runtime__.registerModule("hmr.js", { exports: hmr_exports });
+const foo = "hello";
+text$1(".hmr", foo);
+function text$1(el, text$2) {
+	document.querySelector(el).textContent = text$2;
+}
+hmr_hot?.accept((mod) => {
+	if (mod) {
+		text$1(".hmr", mod.foo);
+	}
+});
+
+//#endregion
+//#region main.js
+var main_exports = {};
+const main_hot = __rolldown_runtime__.createModuleHotContext("main.js");
+__rolldown_runtime__.registerModule("main.js", { exports: main_exports });
+text(".app", "hello");
+function text(el, text$2) {
+	document.querySelector(el).textContent = text$2;
+}
+
+//#endregion
+```
+
+# HMR Step 0
+
+## HMR Update 0
+
+### Code
+
+```js
+//#region hmr.js
+var init_hmr_0 = __rolldown_runtime__.createEsmInitializer((function() {
+	try {
+		var __rolldown_exports__ = __rolldown_runtime__.__export({ foo: () => foo });
+		__rolldown_runtime__.registerModule("hmr.js", { exports: __rolldown_exports__ });
+		const hot_hmr = __rolldown_runtime__.createModuleHotContext("hmr.js");
+		const foo = "hello1";
+		text(".hmr", foo);
+		function text(el, text) {
+			document.querySelector(el).textContent = text;
+		}
+		hot_hmr?.accept((mod) => {
+			if (mod) {
+				text(".hmr", mod.foo);
+			}
+		});
+	} finally {}
+}));
+
+//#endregion
+init_hmr_0()
+__rolldown_runtime__.applyUpdates([['hmr.js', 'hmr.js']]);
+```
+
+### Meta
+
+- update type: patch
+
+#### Hmr Boundaries
+
+- boundary: hmr.js, accepted_via: hmr.js
+
+## HMR Update 1
+
+### Code
+
+```js
+//#region hmr.js
+var init_hmr_0 = __rolldown_runtime__.createEsmInitializer((function() {
+	try {
+		var __rolldown_exports__ = __rolldown_runtime__.__export({ foo: () => foo });
+		__rolldown_runtime__.registerModule("hmr.js", { exports: __rolldown_exports__ });
+		const hot_hmr = __rolldown_runtime__.createModuleHotContext("hmr.js");
+		const foo = "hello2";
```

Comment:
@sapphi-red It doesn't seem to hang within pure rust logic for the case about triggering hmr update during generating hmr update. [Test logic](https://github.com/rolldown/rolldown/pull/7249/files#r2574474634).